client/src/resources/amount.rs

```diff
@@ -197,3 +198,86 @@ mod deserialize_amount_tests {
         assert!(amount.is_err());
     }
 }
+
+/// Recreating ParseIntError since its interior is private
+/// https://doc.rust-lang.org/src/core/num/mod.rs.html#3949
+#[derive(Debug)]
+pub enum ParseIntError {
+    Empty,
+    InvalidDigit,
+    Overflow,
+    Underflow,
+}
+
+// There is a built in __description() method that gives
+// the error as a string, however that is only availible on
+// nightly...
+// So I went with @amosbatto's hack here:
+// https://github.com/rust-lang/rust/issues/22639
+impl From<num::ParseIntError> for ParseIntError {
+    fn from(error: num::ParseIntError) -> Self {
+        match error.to_string().as_ref() {
+            "cannot parse integer from empty string" => ParseIntError::Empty,
```

Comment:
These may change with different versions of rust so I wouldn't rely on them for matching if possible. If you want to convert I would actually just wrap the ParseIntError:

```rust
pub enum ParseAmountError {
    IntError(ParseIntError),
}
```

Then the from implementation is easier:

```rust
impl From<num::ParseIntError> for ParseAmountError {
    fn from(error: num::ParseIntError) -> ParseAmountError {
        ParseAmountError::IntError(error)
    }
}
```

If desired, you can implement `Error` trait on this if you want to output the inner description. But It should capture the data you need.

Reply:
Oh yes, thanks good point. I guess the reason I created the entire matching was because I was unable to figure out how to properly throw an error here:

```rust
if number_decimals > 7 {
                    Err(ParseIntError::Underflow)
                }
```

The reason is that the ParseIntError's value is private so I couldn't instantiate/create a new ParseIntError.